maven-release-manager/src/main/java/org/apache/maven/shared/release/phase/AbstractRewritePomsPhase.java

```diff
@@ -89,6 +93,18 @@ public abstract class AbstractRewritePomsPhase extends AbstractReleasePhase impl
      */
     private String modelETL = JDomModelETLFactory.NAME;
 
+    /**
+     * Regular expression pattern matching Maven expressions (i.e. references to Maven properties).
+     * The first group selects the property name the expression refers to.
+     */
+    private static final Pattern EXPRESSION_PATTERN = Pattern.compile("\\$\\{(.+)\\}");
+
+    /**
+     * All Maven properties allowed to be referenced in parent versions via expressions
+     * @see <a href="https://maven.apache.org/maven-ci-friendly.html">CI-Friendly Versions</a>
+     */
+    private static final List<String> CI_FRIENDLY_PROPERTIES = Arrays.asList("revision", "sha1", "changelist");
```

Comment:
Was this copied from Core? I remember to have this seen this.

Reply:
It is available in https://github.com/apache/maven/blob/eff2fbf2a9baaa2178825b5745c7f152330c8946/maven-model-builder/src/main/java/org/apache/maven/model/interpolation/DefaultModelVersionProcessor.java#L38 but it has only been introduced with 3.8.5 (https://issues.apache.org/jira/browse/MNG-7407) so I would stick with the separate copy inside this class.

Reply:
That's exactly the answer I wanted to hear!